frontend/src/metabase-lib/queries/utils/dataset.ts

```diff
@@ -31,27 +32,56 @@ export function normalizeFieldRef(fieldRef: FieldReference) {
 export function findColumnIndexForColumnSetting(
   columns: DatasetColumn[],
   columnSetting: TableColumnOrderSetting,
+  query?: Lib.Query,
 ) {
   const fieldRef = columnSetting.fieldRef;
   // NOTE: need to normalize field refs because they may be old style [fk->, 1, 2]
+  // TODO: use import { normalize } from "metabase-lib/queries/utils/normalize";
```

Comment:
Is this going to be handled here or in a subtask?

Reply:
yes, I'll add it to the cleanup task